src/EnergyPlus/CurveManager.cc

```diff
@@ -6415,6 +6471,439 @@ Label999: ;
 
 	}
 
+	int
+	GetCurveInterpolationMethodNum( int const CurveIndex ) // index of curve in curve array
+	{
+
+		// PURPOSE OF THIS FUNCTION:
+		// get the interpolation type integer identifier for tables
+
+		// Return value
+		int TableInterpolationMethodNum;
+
+		if ( CurveIndex > 0 ) {
+			TableInterpolationMethodNum = PerfCurve( CurveIndex ).InterpolationType;
+		} else {
+			TableInterpolationMethodNum = 0;
+		}
+
+		return TableInterpolationMethodNum;
+	}
+
+	void
+	ReadTwoVarTableDataFromFile(
+		int const CurveNum,
+		std::string & FileName,
+		int & lineNum
+	)
+	{
+
+		// PURPOSE OF THIS FUNCTION:
+		// get data from an external file used to retrieve optical properties
+
+		// METHODOLOGY EMPLOYED:
+		// Recommended by NREL to read data
+
+		// Using/Aliasing
+		using DataSystemVariables::CheckForActualFileName;
+		using DataSystemVariables::TempFullFileName;
+		using General::splitString;
+
+		int TableNum;
+		bool FileExists;
+		std::ifstream infile( FileName );
+		std::string line;
+
+		CheckForActualFileName( FileName, FileExists, TempFullFileName );
+		if ( !FileExists ) {
+			ShowSevereError( "CurveManager::ReadTwoVarTableDataFromFile: Could not open Table Data File, expecting it as file name = " + FileName );
+			ShowContinueError( "Certain run environments require a full path to be included with the file name in the input field." );
+			ShowContinueError( "Try again with putting full path and file name in the field." );
+			ShowFatalError( "Program terminates due to these conditions." );
+		}
+
+		TableNum = PerfCurve( CurveNum ).TableIndex;
```

Comment:
Thanks for find the problem. Here is my understanding:
1. Remove the line "		std::ifstream infile( FileName );"
2. Replace "infile" by TempFullFileName in the line below
		while (std::getline( infile, line )) {
Please let me know if these actions are OK.